### ISSUE TITLE
chifra export discrepancies between docs and actuality in returned data

### ISSUE THREAD

**Issue by wschwab** [TrueBlocks/trueblocks-core#2546]
Before I get started:
```bash
% chifra --version
chifra version GHC-TrueBlocks//0.40.2-beta
```
## Transactions

If I'm reading the [docs](https://trueblocks.io/data-model/chaindata/#transaction) right, here's the data fields I would expect from running `chifra export` on an address:

Field |
---|
hash
blockHash
blockNumber
transactionIndex
nonce
timestamp
from
to
value
gas
gasPrice
input
receipt
statements
articulatedTx
compressedTx
hasToken
finalized
extraData
isError
date

This is the header file from the csv when I actually run `chifra export`:
```
"blockNumber","transactionIndex","date","timestamp","from","to","ether","ethGasPrice","gasUsed","hash","isError","encoding","compressedTx"
```
The order is quite different, and there are both things in the docs that aren't there (eg nonce) and things there that aren't in the docs (eg encoding). I get the same header if I run `chifra export` with the `-a` flag, but I've raised a separate issue for that (#2545 ).

## Logs

The same goes for `chifra export --logs`. According to the [docs](https://trueblocks.io/data-model/chaindata/#log) I should be getting:

Field |
---|
blockNumber
transactionIndex
logIndex
transactionHash
timestamp
address
topics
data
articulatedLog
compressedLog

What I get:
```
"blockNumber","transactionIndex","logIndex","timestamp","address","topic0","topic1","topic2","topic3","data","compressedLog"
```
The topics aren't in an array, but rather separate values, and the transactionHash isn't present.

## Traces

I'm a bit confused about the docs for traces. There are a few sections: [Trace](https://trueblocks.io/data-model/chaindata/#trace), [TraceAction](https://trueblocks.io/data-model/chaindata/#traceaction), and [TraceResult](https://trueblocks.io/data-model/chaindata/#traceresult). Forgive me, but I'm a bit lazy about copying all of them into tables here, so I'm going to just paste what I _am_ getting when I run `chifra export --traces`:
```
"blockNumber","transactionIndex","traceAddress","action::calltype","error","action::from","action::to","action::value","action::ether","action::gas","result::gasUsed","action::input","compressedTrace","result::output"
```
This seems to be made up of some fields from all three (Trace, TraceAction, and TraceResult), but not all of any of them.

## In Conclusion

I suspect I might just be understanding the docs wrong. 🙂

**Comment by tjayrush** [comment]
Thank you so much for doing this. It might take a while to close this (as it feels like a documentation issue), but it will eventually get closed. I promise.

**Comment by tjayrush** [comment]
Just a note to myself. I think part of this issue is that the fields described in the documentation are for the JSON export. In some cases (this is one), the export to `csv` or `txt` removes a few fields. Both formats should be documented.

**Comment by wschwab** [comment]
I'd argue all formats should return the same data, no?

**Comment by tjayrush** [comment]
Maybe. I've always felt that `--fmt txt` and `--fmt csv` can be identical and `--fmt txt --verbose`, `--fmt csv --verbose` and `--fmt json` could have the full data. Otherwise, the number of columns in `csv` and `txt` can get almost un-managably large.

Having said this, even this isn't how it works. It's a bit of a mess.

We're going to try to clean it up as we move to pure GoLang.

**Comment by tjayrush** [comment]
Such a great issue. Thanks @wschwab We'll be working on this soon (tm).

**Comment by tjayrush** [comment]
**Blocks:**

Add unclesCnt when exporting to csv or txt
Document that transactions, uncles, and withdrawals show cnt when exporting to csv and txt
Document gasUsed

**Comment by tjayrush** [comment]
**Transactions**

_json only_
- articulatedTx, blockHash, value, gas, gasPrice, input, nonce, receipt, traces

_json only, not documented_
- compressedTx, encoding, maxFeePerGas, maxPriorityFeePerGas

_not documented_
- ether, gasCost, gasUsed, type

_txt only_
- isError

_accounting only_
- hasToken, statements

**Comment by tjayrush** [comment]
**Withdrawals**

All documented, all included in both json and txt

**Comment by tjayrush** [comment]
**Receipt**

_only documented_
contractAddress, isError, blockHash

_missing from txt_
gasUsed, logs, status, transactionHash, transactionIndex

_missing timestamp and date_

**Comment by tjayrush** [comment]
**Logs**

_txt only_
topic0, topic1, topic2, topic3

_not in txt_
topics

_txt only with --articulatre_
compressedLog

_json only with artciculate_
articulatedLog

**Comment by tjayrush** [comment]
**Trace**

_not in txt_
action, articulatedTrace, blockHash, date, subtraces, traceAddress, transactionHash, type

_not in json_
compressedTrace, date

_not documented, not json_
error

**Comment by tjayrush** [comment]
**Trace Action**

_txt only_
ether

_documented only_
refundAddress

_documented only_
rewardType

_not documented_
value

**Comment by tjayrush** [comment]
**Trace Result**

_documented only_
address, code

**Comment by tjayrush** [comment]
**Appearances**

Json
CSV/Txt
Docs

**Comment by tjayrush** [comment]
**Neighbors**

Json
CSV/Txt
Docs

**Comment by tjayrush** [comment]
**Accounting**

Json
CSV/Txt
Docs

**Comment by tjayrush** [comment]
**Statements**

Json
CSV/Txt
Docs

**Comment by tjayrush** [comment]
**Balances**

Json
CSV/Txt
Docs

**Comment by tjayrush** [comment]
**Count**

Json
CSV/Txt
Docs

**Comment by tjayrush** [comment]
I am going to close this. It's now well documented, and when our new auto-code generation tools improve, this issue will resolve naturally.
